source/dub/internal/utils.d

```diff
@@ -40,8 +40,14 @@ NativePath getTempDir()
 NativePath getTempFile(string prefix, string extension = null)
 {
 	import std.uuid : randomUUID;
+    import std.array: replace;
 
-	auto path = getTempDir() ~ (prefix ~ "-" ~ randomUUID.toString() ~ extension);
+	string fileName = prefix ~ "-" ~ randomUUID.toString() ~ extension;
+
+	if (extension !is null && extension == ".d")
+		fileName = fileName.replace("-", "_");
```

Comment:
Maybe directly use _ above?

Reply:
randomUUID also returns a string containing dashes. Therefore my idea was to keep the dash and replaces all dashes in case of a d module file.